torch/csrc/distributed/autograd/context/dist_autograd_context.cpp

```diff
@@ -12,6 +12,17 @@ int64_t DistAutogradContext::context_id() const {
   return context_id_;
 }
 
+std::vector<rpc::WorkerId> DistAutogradContext::getKnownWorkerIds() const {
+  return knownWorkerIDs_;
+};
+void DistAutogradContext::addKnownWorkerID(const rpc::WorkerId& workerId) {
+  std::lock_guard<std::mutex> guard(lock_);
+  if (std::find(knownWorkerIDs_.begin(), knownWorkerIDs_.end(), workerId) ==
+      knownWorkerIDs_.end()) {
+    knownWorkerIDs_.push_back(workerId);
+  }
+}
+
```

Comment:
This is O(N) and the insertion check happens every call. Instead, you could use an `unordered_set`, or even a bitset, given that the range is only `uint16_t`. This means this set would only use 8KB of memory and you get to set a bit on every call.